src/plugins/embeddable/common/lib/migrate.ts

```diff
@@ -26,9 +26,12 @@ export const getMigrateFunction = (embeddables: CommonEmbeddableStartContract) =
     updatedInput.enhancements = {};
     Object.keys(enhancements).forEach((key) => {
       if (!enhancements[key]) return;
-      (updatedInput.enhancements! as Record<string, any>)[key] = embeddables
-        .getEnhancement(key)
-        .migrations[version](enhancements[key] as SerializableState);
+      const enhancementDefinition = embeddables.getEnhancement(key);
+      if (enhancementDefinition.migrations && enhancementDefinition.migrations[version]) {
+        (updatedInput.enhancements! as Record<string, any>)[key] = enhancementDefinition.migrations[
+          version
+        ](enhancements[key] as SerializableState);
+      }
```

Comment:
I tested this function. It _should_ run migrations on enhancements which have them, and retain enhancements which don't.

 ```suggestion
      const enhancementDefinition = embeddables.getEnhancement(key);
      const migratedEnhancement = enhancementDefinition?.migrations?.[version]
        ? enhancementDefinition.migrations[version](enhancements[key] as SerializableState)
        : enhancements[key];
      (updatedInput.enhancements! as Record<string, any>)[key] = migratedEnhancement;
```

Reply:
@ppisljar - would you be able to add a test that would have caught this issue? Unless I'm missing it, it looks like the only test added was for the original bug, not the secondary bug.

Reply:
the test was added to catch this as well

Reply:
Great, thanks Peter!

Reply:
Actually, did you verify this? I just copied the old code into the file and ran tests, and they all still pass.

<img width="769" alt="Screen Shot 2021-06-15 at 7 37 43 AM" src="https://user-images.githubusercontent.com/16563603/122046230-a4547700-cdac-11eb-8890-93cee5e08fb2.png">

Reply:
original code should fail the test as there is no migration defined for 7.10.0 

and about the case you are asking about (this thread), original code didn't show that bug, so from that perspective its ok for the tests to pass.

Reply:
It's my understanding (maybe I'm wrong), that your first fix had a bug that Devon found and it wasn't caught by tests?  That is the situation I am asking for tests for. Just because the original code didn't show that bug doesn't mean there shouldn't be a test for it.

Reply:
because original code didn't show that bug you trying to take original code and get the new test failing doesn't work.
you would need to take the intermediate code (where devon found a bug) and then the test would fail

Reply:
I synced with @ppisljar about this, and we saw how if we take the first iteration of code (pasted below), and run the tests, they all still pass:

```
  const enhancementDefinition = embeddables.getEnhancement(key);
      if (enhancementDefinition.migrations && enhancementDefinition.migrations[version]) {
        (updatedInput.enhancements! as Record<string, any>)[key] = enhancementDefinition.migrations[
          version
        ](enhancements[key] as SerializableState);
      }
```

@ppisljar - can you follow up when you've added a test that would catch the bug?

Reply:
This definitely needs unit test coverage. 

As some _functional_ coverage for this issue, In [this PR](https://github.com/elastic/kibana/pull/102667) I assert that all drilldowns survive the migration process (on by value & by reference panels). The original code removed drilldowns from by value panels, so if we run into something like this again, the smoke test should fail.

Reply:
Filed https://github.com/elastic/kibana/issues/102772 to keep track